application_sandbox/overlapping_frames/main.cpp

```diff
@@ -187,515 +219,460 @@ vulkan::VulkanGraphicsPipeline buildTrianglePipeline(
+        1,                              // mipLevels
+        1,                              // arrayLayers
+        VK_SAMPLE_COUNT_1_BIT,          // samples
+        VK_IMAGE_TILING_OPTIMAL,        // tiling
+        VK_IMAGE_USAGE_SAMPLED_BIT |
+            VK_IMAGE_USAGE_COLOR_ATTACHMENT_BIT,  // usage
+        VK_SHARING_MODE_EXCLUSIVE,                // sharingMode
+        0,                                        // queueFamilyIndexCount
+        nullptr,                                  // pQueueFamilyIndices
+        VK_IMAGE_LAYOUT_UNDEFINED                 // initialLayout
+    };
+    images.push_back(app->CreateAndBindImage(&image_create_info));
+  }
+
+  return images;
 }
 
 containers::vector<vulkan::VkImageView> buildSwapchainImageViews(
-    vulkan::VulkanApplication *app,
-    const entry::EntryData* data) {
-    containers::vector<vulkan::VkImageView> image_views(data->allocator());
-    image_views.reserve(app->swapchain_images().size());
-
-    for(int index=0; index < app->swapchain_images().size(); index++) {
-        VkImage& swapchain_image = app->swapchain_images()[index];
-
-        VkImageViewCreateInfo image_view_create_info {
-            VK_STRUCTURE_TYPE_IMAGE_VIEW_CREATE_INFO,  // sType
-            nullptr,                                   // pNext
-            0,                                         // flags
-            swapchain_image,                           // image
-            VK_IMAGE_VIEW_TYPE_2D,                     // viewType
-            app->swapchain().format(),                 // format
-            {
-                VK_COMPONENT_SWIZZLE_IDENTITY,  // components.r
-                VK_COMPONENT_SWIZZLE_IDENTITY,  // components.g
-                VK_COMPONENT_SWIZZLE_IDENTITY,  // components.b
-                VK_COMPONENT_SWIZZLE_IDENTITY,  // components.a
-            },
-            {
-                VK_IMAGE_ASPECT_COLOR_BIT,  // subresourceRange.aspectMask
-                0,                          // subresourceRange.baseMipLevel
-                1,                          // subresourceRange.levelCount
-                0,                          // subresourceRange.baseArrayLayer
-                1,                          // subresourceRange.layerCount
-            },
-        };
-
-        VkImageView raw_image_view;
-        LOG_ASSERT(==, data->logger(), app->device()->vkCreateImageView(
-            app->device(),
-            &image_view_create_info,
-            nullptr,
-            &raw_image_view
-        ), VK_SUCCESS);
-        image_views.push_back(vulkan::VkImageView(raw_image_view, nullptr, &app->device()));
-    }
-
-    return image_views;
+    vulkan::VulkanApplication* app, const entry::EntryData* data) {
+  containers::vector<vulkan::VkImageView> image_views(data->allocator());
+  image_views.reserve(app->swapchain_images().size());
+
+  for (int index = 0; index < app->swapchain_images().size(); index++) {
```

Comment:
While we are making changes, can we switch this in 2 ways, 
1) size_t (since we are comparing with swapchain_images.size(), which is a size_t). Prevents a possible(unlikely) infinite loop
2) `size_t i = 0` to be more idiomatic C++

Reply:
done